subprojects/studio-plugin/src/main/java/org/gradle/profiler/studio/plugin/system/AndroidStudioSystemHelper.java

```diff
@@ -35,38 +43,54 @@ public static GradleSyncResult doGradleSync(Project project) {
      * Registers a listener that waits on next gradle sync if it's in progress.
      */
     public static void waitOnPreviousGradleSyncFinish(Project project) {
-        GradleProfilerGradleSyncListener syncListener = new GradleProfilerGradleSyncListener();
-        MessageBusConnection connection = GradleSyncState.subscribe(project, syncListener);
-        if (!GradleSyncState.getInstance(project).isSyncInProgress()) {
+        MessageBusConnection connection = project.getMessageBus().connect();
+        CompletableFuture<String> future = new CompletableFuture<>();
+        connection.subscribe(PROJECT_SYSTEM_SYNC_TOPIC, syncResult -> future.complete("Done"));
```

Comment:
Shouldn't this only happen after checking if a sync is already in progress?

Reply:
We can also do it that way, changed